src/main/java/org/kohsuke/github/GHCheckRunBuilder.java

```diff
@@ -0,0 +1,358 @@
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Drafts a check run.
+ *
+ * @see GHCheckRun
+ * @see GHRepository#createCheckRun
+ * @see <a href="https://developer.github.com/v3/checks/runs/">documentation</a>
+ */
+@SuppressFBWarnings(value = "URF_UNREAD_FIELD", justification = "Jackson serializes these even without a getter")
+@Preview
+@Deprecated
+public final class GHCheckRunBuilder {
+
+    private final GHRepository repo;
+    private final Requester requester;
+    DraftOutput output;
+    List<DraftAction> actions;
+
+    GHCheckRunBuilder(GHRepository repo, String name, String headSHA) {
+        this.repo = repo;
+        requester = repo.root.createRequest()
+                .withPreview(Previews.ANTIOPE)
+                .method("POST")
+                .with("name", name)
+                .with("head_sha", headSHA)
+                .withUrlPath(repo.getApiTailUrl("check-runs"));
+    }
+
+    public @NonNull GHCheckRunBuilder withDetailsURL(@CheckForNull String detailsURL) {
+        requester.with("details_url", detailsURL);
+        return this;
+    }
+
+    public @NonNull GHCheckRunBuilder withExternalID(@CheckForNull String externalID) {
+        requester.with("external_id", externalID);
+        return this;
+    }
+
+    public @NonNull GHCheckRunBuilder withStatus(@CheckForNull GHCheckRun.Status status) {
+        if (status != null) {
+            // Do *not* use the overload taking Enum, as that s/_/-/g which would be wrong here.
+            requester.with("status", status.toString().toLowerCase(Locale.ROOT));
+        }
+        return this;
+    }
+
+    public @NonNull GHCheckRunBuilder withConclusion(@CheckForNull GHCheckRun.Conclusion conclusion) {
+        if (conclusion != null) {
+            requester.with("conclusion", conclusion.toString().toLowerCase(Locale.ROOT));
+        }
+        return this;
+    }
+
+    public @NonNull GHCheckRunBuilder withStartedAt(@CheckForNull Date startedAt) {
+        if (startedAt != null) {
+            requester.with("started_at", GitHubClient.printDate(startedAt));
+        }
+        return this;
+    }
+
+    public @NonNull GHCheckRunBuilder withCompletedAt(@CheckForNull Date completedAt) {
+        if (completedAt != null) {
+            requester.with("completed_at", GitHubClient.printDate(completedAt));
+        }
+        return this;
+    }
+
+    /**
+     * Drafts the output section.
+     *
+     * @param title
+     *            as in GitHub documentation
+     * @param summary
+     *            as in GitHub documentation
+     * @return use {@link DraftOutput#done} to continue
+     */
+    public @NonNull DraftOutput withOutput(@NonNull String title, @NonNull String summary) {
+        return new DraftOutput(this, title, summary);
+    }
+
+    /**
+     * Drafts an action section.
```

Comment:
Add links to the pertinent GitHub docs, such as https://developer.github.com/v3/checks/runs/#actions-object.

Reply:
There's a maximum of three actions allowed.  But the GitHub docs do not make it clear what happens when actions are exceeded.  We should at least mention the limit here.

Reply:
As elsewhere, I think it is pointless to even mention these sorts of limitations in Javadoc here, as the limits could be changed at any time and the code in this repository would become stale and misleading. (For example, some fields accept quite arbitrary-seeming lengths such as 20 characters.) It is the caller’s responsibility to follow GitHub’s current limits, and deal with whatever errors result if those are exceeded.

Reply:
(The exception being the 50 annotation limit, since that has a documented and tested solution—adding annotations in batches of 50 or less—which can be implemented in the library.)